### DISCUSSION TITLE
Direct Installation of Yarn 3.x?

### DISCUSSION THREAD

**Discussion by ghost** [yarnpkg/berry#5118] [Q&A] [answered]
I have been assisting on a few projects that are still using Yarn 1.22.19. I was able to look at the GitHub Actions logs for a particular repository, and I noticed that tests on MacOS and Ubuntu/Linux were not performing well, and tests on Windows were failing. I have a few questions about the installation and deprecation process:

1. When installing Yarn, why is the latest version not installed from the get-go? It would greatly simplify the startup process because the version of Yarn does not need to be set manually.
2. Why has v1.x not been deprecated or removed yet? Not doing so can lead to confusion among teams and contributors and can cause new contributors to stumble and end up hunting for bugs when they do not know they are using the wrong version.

**Comment by arcanis** [comment, marked as answer]
> When installing Yarn, why is the latest version not installed from the get-go? It would greatly simplify the startup process because the version of Yarn does not need to be set manually.

https://github.com/yarnpkg/berry/issues/5119#issuecomment-1340735406

We don't have a lot to gain from this, and a lot to lose. As far as we know many things still rely on Yarn 1 being the global, if only because of the different CLI commands and options, so we'd risk breaking many CIs.

> Why has v1.x not been deprecated or removed yet? Not doing so can lead to confusion among teams and contributors and can cause new contributors to stumble and end up hunting for bugs when they do not know they are using the wrong version.

Because it works fine for projects that already use it. It has bugs, and it's not the best we have today, but deprecating it would create "panic", with everyone thinking they HAVE to migrate away from it when they aren't forced at all. We'll probably be pushing people to upgrade a bit more in the next years, but there's no plan to fully deprecate v1 - we'll still release security upgrades there, if we have to.
